video_creation/screenshot_downloader.py

```diff
@@ -22,27 +21,53 @@ def download_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: in
         reddit_object (Dict): Reddit object received from reddit/subreddit.py
         screenshot_num (int): Number of screenshots to download
     """
+    # settings values
+    W: Final[int] = int(settings.config["settings"]["resolution_w"])
+    H: Final[int] = int(settings.config["settings"]["resolution_h"])
+    lang: Final[str] = settings.config["reddit"]["thread"]["post_lang"]
+    storymode: Final[bool] = settings.config["settings"]["storymode"]
+
     print_step("Downloading screenshots of reddit posts...")
-    id = re.sub(r"[^\w\s-]", "", reddit_object["thread_id"])
+    reddit_id = re.sub(r"[^\w\s-]", "", reddit_object["thread_id"])
     # ! Make sure the reddit screenshots folder exists
-    Path(f"assets/temp/{id}/png").mkdir(parents=True, exist_ok=True)
+    Path(f"assets/temp/{reddit_id}/png").mkdir(parents=True, exist_ok=True)
 
     with sync_playwright() as p:
         print_substep("Launching Headless Browser...")
 
         browser = p.chromium.launch(headless=True)  # add headless=False for debug
-        context = browser.new_context()
 
+        # Device scale factor (or dsf for short) allows us to increase the resolution of the screenshots
+        # When the dsf is 1, the width of the screenshot is 600 pixels
+        # so we need a dsf such that the width of the screenshot is greater than the final resolution of the video
+        dsf = (W // 600) + 1
+
+        context = browser.new_context(
+            locale=lang or "en-us",
```

Comment:
Is this for the translation of the buttons in the screenshots?

Reply:
Yes, but I still need to test it. I think it should work if the lang is an ISO code.

Reply:
Nice, maybe we need to add a check for it

Reply:
Okay I tested it with lang=it and it works.